src/views/CityExplorer/CityExplorerMain.js

```diff
@@ -0,0 +1,65 @@
+import React, { Component } from 'react';
+import Map from './Map';
+import Weather from './Weather';
+import Restaurants from './Restaurants';
+import location from '../../assets/data/location.json';
+import restaurants from '../../assets/data/restaurant.json';
+import weather from '../../assets/data/weather.json';
+import map from '../../assets/Images/map.png';
+
+export class CityExplorerMain extends Component {
+  
+constructor(props) {
+  super(props)
+  this.state = {
+     displaySearchResults: false,
+     locationObject: location,
+     weather: weather,
+     restaurants: restaurants
+  }
+}
+
+handleLocationSearch = (event) => {
+  event.preventDefault();
+  console.log(location, restaurants, weather);
+  /** add set state to update view after search */
+  this.setState({displaySearchResults: true});
+}
+
+
+  render() {
+    return (
+      
+      <div>
+      <div id="header">
+      <h1>Find Your Path</h1>
+      <p>Enter a location in the USA below to learn about the weather, restaurants, movies, and more!</p>
+    </div>
+        <form onSubmit={this.handleLocationSearch}>
+          <label>Find the Fun.</label>
+          <input type="text"  name="search" placeholder="Enter a location." />
+          <button type="sumbit">Find It!</button>
+        </form>
+        {/**add condtional rendering for search */}
+        { this.state.displaySearchResults &&
```

Comment:
I've noticed you used this syntax in several places (across multiple files) and am wondering how this works? I'm not familiar with this syntax.

Reply:
I am guessing you are referring to the {} and the conditional rendering of the component view based on if state is true and the user defined components, restaurant, location, map? The {} allow for js and I think I need them for the comment as well in the render. @selmaT273

Reply:
https://reactjs.org/docs/conditional-rendering.html#inline-if-with-logical--operator

Reply:
This we need to make sure that students fully understand, condition ? true : false.

Reply:
Ah, ok. I see that the React docs in that link has the `&&` operator in an example for conditionally including an element/component, but now I am wondering if it is standard practice in React to use this syntax for an if condition, as opposed to using a ternary? 

I agree, the students need to understand how to use ternary syntax (condition ? true : false) as it is specified in the curriculum, but I don't believe the way its written here on lines 43-60 is an example of a ternary as there is no true or false, it only states the condition (missing colon and question mark). I can see the conditional syntax used in these lines as potentially being confusing for newer developers.

Reply:
Although hopefully the students are not looking at my code.

Reply:
If you prefer to use an inline if with && instead of ternary, I'm definitely up for chatting more about why and think it could be useful to show students both ways. And yes while I don't necessarily think they'll look at this repo, I was moreso thinking the code you're writing out here is reflective of your preference and/or what you plan to loosely base your demo off of.

Reply:
To the original question: `condition && <MyComponent />` is a super common React idiom to only render `MyComponent` if `condition` is truthy, i.e. only render search results if there has been a search query. This is exactly how I would have written it.

Reinforcing the ternary conditional (`condition ? ifTrue : ifFalse`) is _also_ important, and the two syntaxes are closely related. The `&&` above is also identical to `condition ? <MyComponent /> : condition`, i.e. the value of the expression is `condition` if `condition` evaluates as falsey (false, null, undefined, 0, ""). The quirk is React chooses not to render some falsey things (`false`, `null`, `""`), renders others (`0`), and complains about `undefined`. The [React Conditional Rendering link](https://reactjs.org/docs/conditional-rendering.html#inline-if-with-logical--operator) includes a section on how you get `<div>0</div>` due to this, when what you probably wanted is not to show a 0.

All of this is stuff we need to understand as instructors, and then choose how deep down the rabbit hole to go in explaining to the class. I've definitely "accidentally" had a 0 show up on the page before with `&&` to clarify that particular quirk, which isn't necessarily obvious.

TL;DR: React devs tend to take advantage of how `||` and `&&` and `? ... :` work in JS, and this seems like a good lecture/demo to reinforce all of that.

> Although hopefully the students are not looking at my code.

They might! Regardless, you should both be practicing how you would "perform", i.e. good variable names, indentation, component decomposition, etc.

Reply:
So the <div>0</div> rendering is the return of the false condition being rendered and react chooses 0 as the value? @dahlbyk

Reply:
> So the `<div>0</div> ` rendering is the return of the false condition being rendered and react chooses 0 as the value? @dahlbyk

I wouldn't say React "chooses 0 as the value". It's the value because that's how JS works, and then React decides if/how to render the value.

Two steps:
1. JS evaluates the `&&`, which results in the left expression if it's falsey (e.g. 0).
2. React renders the result of the expression in `{...}`, which is `0`. Since 0 is a useful thing to display to the screen, React renders the 0 inside the `<div>`.
    - React _will not_ render `false` or `null` values, which are also falsey. This is really just a quirk of `0` being falsey but still rendered by React.

Reply:
so does the 0 come from the empty object being returned, or from the first half of the condition where the length did not equate to greater than 0? @dahlbyk

Reply:
or is that just JS and react being react?